packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -184,12 +186,45 @@ export function toCollectorSpan(
     droppedAttributesCount: 0,
     events: toCollectorEvents(span.events),
     droppedEventsCount: 0,
-    status: span.status,
+    status: toCollectorStatus(span.status),
     links: toCollectorLinks(span, useHex),
     droppedLinksCount: 0,
   };
 }
 
+/**
+ * Converts StatusCode
+ * @param code
+ */
+export function toCollectorCode(
+  code: StatusCode
+): opentelemetryProto.trace.v1.StatusCode {
+  switch (code) {
+    case 0:
+      return 1;
+    case 1:
+      return 0;
+    default:
+      return 2;
+  }
+}
+
+/**
+ * Converts status
+ * @param status
+ */
+export function toCollectorStatus(
+  status: Status
+): opentelemetryProto.trace.v1.Status {
+  const spanStatus: opentelemetryProto.trace.v1.Status = {
+    code: toCollectorCode(status.code),
+  };
+  if (typeof status.message !== 'undefined') {
```

Comment:
I think it would be better to directly test for undefined value instead of using `typeof`
```js
if(status.message !== undefined)
```

Reply:
not really, if message is set to anything I just pass it through, this is not validation but just checking if the message was set or not and then pass it through exactly as it was

Reply:
Yes but to check it you are using `typeof` and then compare the result to a string `'undefined'`.
What is the benefit of it over testing directly for equality to undefined?

Reply:
Unless I am missing something, the current implementation calls the typeof operator and then does string comparison, which is less performant